src/Plugin/formatter/Formatter.php

```diff
@@ -41,4 +44,20 @@ public function getContentTypeHeader() {
     return 'application/hal+json; charset=utf-8';
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function getResource() {
+    if (isset($this->resource)) {
+      return $this->resource;
+    }
+
+    // Get the resource from the instance configuration.
+    $instance_configuration = $this->getConfiguration();
+    if (empty($instance_configuration['resource'])) {
+      return NULL;
+    }
+    return $instance_configuration['resource'] instanceof ResourceInterface ? $instance_configuration['resource'] : NULL;
```

Comment:
This is just type checking. If you use `setResource('ha!')` then `getResource()` will yield `NULL` and not `ha!`. This can go away and let the implementor mess with their own code.